File: CodeChef/Self Learning/Basic Programming Concepts/Min_No_of_Pizzas.cpp
// https://www.codechef.com/LP2TO301/problems/MINPIZZAS
#include<bits/stdc++.h>
using namespace std;

int main(){
    
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int t;
    cin>>t;

    while(t--)
    {
      int N,K;
      cin>>N>>K;      
       //cout<<(std::lcm(N,K))/K<<"\n"; //this can also be used
       cout<<N/__gcd(N,K)<<"\n";
    } 

    return 0;
}

File: CodeChef/Self Learning/Basic Programming Concepts/Min_No_of_Pizzas.java
// https://www.codechef.com/LP2TO301/problems/MINPIZZAS
// import java.util.*;

// class Min_No_of_Pizzas {

//   static int gcd(int a,int b){
//     if(b==0) return a;
//     return gcd(b,a%b);
//   }

//     public static void main(String[] args) {
//       Scanner sc=new Scanner(System.in);
       
//        int t=sc.nextInt();
//         while(t>0){
//          int N=sc.nextInt();
//          int K=sc.nextInt();
        
//          int g=2;
//          if(N>K){
//            gcd(N,K);
//          }
//          else if(N<K){
//            g=gcd(K,N);
//          }
//          else g=1;
//          N=N/g;
//          System.out.println(N);

//          t--;
//        }
// }
// }

//tle error 
